src/psyclone/transformations.py

```diff
@@ -2825,3 +2824,216 @@ def validate(self, loop):
+        return "ExtractRegionTrans"
+
+    def _validate(self, node_list):
+        ''' Perform validation checks before applying the transformation
+
+        :param node_list: the list of Node(s) we are checking.
+        :type node_list: list of :py:class:`psyclone.psyGen.Node`.
+        :raises TransformationError: if distributed memory is configured.
+        :raises TransformationError: if transformation is applied to a Node \
+                                     or a list of Nodes which already \
+                                     contain an ExtractNode.
+        :raises TransformationError: if transformation is applied to a \
+                                     Kernel or a BuiltIn call without its \
+                                     parent Loop.
+        :raises TransformationError: if transformation is applied to a Loop \
+                                     without its parent Directive when \
+                                     optimisations are applied.
+        :raises TransformationError: if transformation is applied to an \
+                                     orphaned Directive without its parent \
+                                     Directive.
+        :raises TransformationError: if transformation is applied to a Loop \
+                                     over cells in a colour without its \
+                                     parent Loop over colours in Dynamo0.3 API.
+        :raises TransformationError: if transformation is applied to an \
+                                     inner Loop without its parent outer \
+                                     Loop in GOcean1.0 API.
+        '''
+
+        # First check constraints on Nodes in the node_list common to
+        # all RegionTrans transformations
+        super(ExtractRegionTrans, self)._validate(node_list)
+
+        # Check ExtractRegionTrans specific constraints.
+        from psyclone.extractor import ExtractNode
+        from psyclone.psyGen import Loop, Kern, BuiltIn, Directive, \
+            OMPDoDirective, ACCLoopDirective, OMPParallelDoDirective
+        from psyclone.dynamo0p3 import DynLoop
+        from psyclone.gocean1p0 import GOLoop
+
+        # Extracting distributed memory code is not supported. This
+        # constraint covers the presence of HaloExchange and GlobalSum
+        # classses as they are only generated when distributed memory
+        # is enabled.
+        if Config.get().distributed_memory:
+            raise TransformationError(
+                "Error in {0}: Distributed memory is not supported."
+                .format(str(self.name)))
+
+        # Now check constrains for individual Nodes in node list
+        for node in node_list:
+
+            # Check that an ExtractNode is not already in the node list
+            # marked for extraction. Otherwise we would have an extract
+            # region within another extract region.
+            if node.walk(node.children, ExtractNode) or \
```

Comment:
This check too will be handled by #292

Reply:
It is so I removed the constraint and modified the relevant test.

Reply:
Mini-whoop!